src/test/java/co/cask/hydrator/python/transform/PythonTransformNativeTest.java

```diff
@@ -0,0 +1,307 @@
+
+import java.io.File;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Unit tests for our plugins.
+ */
+public class PythonTransformNativeTest extends HydratorTestBase {
+  private static final ArtifactSummary APP_ARTIFACT = new ArtifactSummary("python-evaluator", "2.2.0-SNAPSHOT");
+  @ClassRule
+  public static final TestConfiguration CONFIG = new TestConfiguration("explore.enabled", false);
+
+  private static final Schema SCHEMA = Schema.recordOf(
+    "data",
+    Schema.Field.of("s", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+    Schema.Field.of("i", Schema.nullableOf(Schema.of(Schema.Type.INT))),
+    Schema.Field.of("l", Schema.nullableOf(Schema.of(Schema.Type.LONG))),
+    Schema.Field.of("f", Schema.nullableOf(Schema.of(Schema.Type.FLOAT))),
+    Schema.Field.of("d", Schema.nullableOf(Schema.of(Schema.Type.DOUBLE))),
+    Schema.Field.of("b", Schema.nullableOf(Schema.of(Schema.Type.BOOLEAN))));
+
+  @BeforeClass
+  public static void setupTestClass() throws Exception {
+    ArtifactId parentArtifact = NamespaceId.DEFAULT.artifact(APP_ARTIFACT.getName(), APP_ARTIFACT.getVersion());
+
+    // add the artifact and mock plugins
+    setupBatchArtifacts(parentArtifact, DataPipelineApp.class);
+
+    // add our plugins artifact with the artifact as its parent.
+    // this will make our plugins available.
+
+    addPluginArtifact(NamespaceId.DEFAULT.artifact("example-plugins", "1.0.0"),
+                      parentArtifact,
+                      ScriptContext.class,
+                      PythonEvaluator.class,
+                      WeakReferenceDelegatorClassLoader.class);
+  }
+
+  @Test
+  public void testPrimitivesDecodedNative() throws Exception {
+    String script = "def transform(record, emitter, context):\n" +
+      "    emitter.emit(record)\n";
+
+    List<StructuredRecord> outputRecords = runPythonEvaluatorPipeline(
+      ImmutableList.of(
+        StructuredRecord.builder(SCHEMA)
+          .set("s", "ab").set("i", 0).set("l", 0L).set("f", 0f).set("d", 0d).set("b", true).build(),
+        StructuredRecord.builder(SCHEMA)
+          .set("s", "xy").set("i", -10).set("l", -10L).set("f", -10f).set("d", -10d).set("b", true).build(),
+        StructuredRecord.builder(SCHEMA)
+          .set("s", "a").set("i", 10).set("l", 10L).set("f", 10f).set("d", 10d).set("b", false).build(),
+        StructuredRecord.builder(SCHEMA)
+          .set("s", "").set("i", 0).set("l", 0L).set("f", 0f).set("d", 0d).set("b", false).build()),
+
+      ImmutableMap.<String, String>of("script", script,
+                                    "executionMode", getExecutionMode(),
+                                    "pythonBinary", "python"));
+
+    Assert.assertEquals(4, outputRecords.size());
+    Assert.assertTrue(outputRecords.stream().anyMatch(obj -> obj.get("s").equals("xy") &&
```

Comment:
why don't you just compare with predefined expected list (in current sample - same as input list)? I mean all similar places.